### ISSUE TITLE
Investigate GSOD redirect error

### ISSUE THREAD

**Issue by SgtCoDFish** [cert-manager/website#952]
#949 was required to revert a renamed file which caused problems when deployed live.

The link would create an infinite redirect, and this persisted even with a clean rebuild after the cache was cleaned. Not sure if it's netlify or next.js. We should find out because I don't currently know how to safely rename a page.

**Comment by SgtCoDFish** [comment]
I think this is actually pretty simple now I look back at it.

### 1: Differently named index file in old site

My migration script for the old cert-manager site, [rutte](https://github.com/SgtCoDFish/rutte) assumed that all "index" files were called `_index.md`, which was true when I wrote the script initially.

However, on 2022-03-24 (before the migration) we merged #867 which added `content/en/docs/contributing/google-season-of-docs/2022/improve-navigation-and-structure/index.md`.

My script thus didn't know this was an index file and didn't rename it to `README.md`. This meant that after the refresh merged the path changed:

```text
from:
/docs/contributing/google-season-of-docs/2022/improve-navigation-and-structure/

to:
/docs/contributing/google-season-of-docs/2022/improve-navigation-and-structure/index/
```

There was no index page configured, so the path without `index/` would be a 404.

**Takeaway**: I didn't realise that Hugo supported both `_index.md` and `index.md`, and since we didn't use `index.md` anywhere until shortly before the refresh merged, I never spotted this. Careful review of the manifest would have shown this but I never even thought to check for this specific case.

Bad luck that we started using `index.md` at this point, too.

### 2: Redirect rule

On 2022-04-14 #917 was raised which highlighted the breaking change. Several commits were then pushed directly to `master` without a PR in an attempt to fix the issue (so I can't link a PR)

One of those commits fixed the issue by redirecting `<...>/improve-navigation-and-structure/` to `<...>/improve-navigation-and-structure/index/` permanently:

https://github.com/cert-manager/website/blame/ed5c332fbb67989078c6bc8e2d94263017bc3c0c/public/_redirects#L160

I was not aware of this change; I was away at the time it was made.

### 3: I try to be helpful

I spotted the "broken" URL randomly and on 2022-04-26 I raised #941 which, among other things, changed the URL back to how it would've been on the old site.

Crucially, when I test this locally it works absolutely fine **because our redirect rules only apply when we deploy and not when running locally**.

**Takeaway**: I should've spotted that this was a potentially breaking change in its right, and should've looked at adding a redirect rule from `/index`. This would've increased the chances of me spotting the new redirect rule which was added previously.

### 4: Reverted

Because of the redirect rule combined with my "helpful" change to change the path back to the old style, a redirect loop occurred but crucially this was not reproducible on a local environment.

Richard spotted the breakage on live and I raised #949 to revert. Until I spotted the new redirect rule today, I was confused as to what had broken.

### TL;DR

1) I thought we only used `_index.md` but just before the refresh we added `index.md`
2) The fix to the above added a new redirect rule I wasn't aware of
3) I tried to fix (1) "properly"
4) This caused a redirect loop.

**Comment by SgtCoDFish** [comment]
>  We should find out because I don't currently know how to safely rename a page.

To follow up on this specifically: Renaming a page actually works exactly as I'd expected all along, and there's nothing to see here :grin:

**Comment by maelvls** [comment]
I have not properly communicated the hot-fixes I made on Friday, my apologies for this, it has created a lot of confusion.

The issue was raised by Enjeck on Saturday 15 April 2020 in https://github.com/cert-manager/website/issues/917. This issue was problematic because cert-manager just had been selected a few hours earlier (during the night from Friday to Saturday). Since I had to chance to get an approval until the next Tuesday due to a bank holiday, and because the only way to "see" whether a fix on `_redirect` would have an impact, it had to be on the master branch. Thus, I chose to commit to the master branch and skip a pull request, and I referenced Enjeck's issue in each of the commits (I realized that I could have self-lgtm in my own pull requests, I didn't have to push to master!)

The three commits in question are:

- ad85bbd
- 7363767
- 5cd7bab
